src/test-env.ts

```diff
@@ -0,0 +1,119 @@
+import { addressUtils, HDNode, unitsUtils } from '@vechain/sdk-core'
+import { ThorClient } from '@vechain/sdk-network'
+import { genesis } from './constants'
+
+const mnemonic = process.env.MNEMONIC
+const nodeUrls = process.env.NODE_URLS
+const privateKeys = process.env.PRIVATE_KEYS
+const networkType = process.env.NETWORK_TYPE
+
+const NetworkTypes = ['solo', 'default-private', 'testnet', 'mainnet'] as const
+type NetworkType = (typeof NetworkTypes)[number]
+
+const testEnv = {
+    get keys(): string[] {
+        if (privateKeys) {
+            return privateKeys.split(',')
+        }
+
+        if (!mnemonic) {
+            throw new Error('No mnemonic provided')
+        }
+        const _mnemonic = mnemonic
+
+        const hdNode = HDNode.fromMnemonic(_mnemonic.split(' '))
+        const keys: any[] = []
+        for (let i = 0; i < 10; i++) {
+            const child = hdNode.derive(i)
+            keys.push(Buffer.from(child.privateKey!).toString('hex'))
+        }
+
+        return keys
+    },
+    get urls(): string[] {
+        if (nodeUrls) {
+            return nodeUrls.split(',')
+        }
+
+        return ['http://localhost:8669']
+    },
+    get type(): NetworkType {
+        if (networkType) {
+            const type = networkType as NetworkType
+            if (!NetworkTypes.includes(type)) {
+                throw new Error('Invalid network type')
+            }
+            return type
+        }
+
+        return 'default-private'
+    },
+}
+
+const validateEnv = async (): Promise<boolean> => {
+    const genesisIds: string[] = []
+
+    if (nodeUrls) {
+        const urls = nodeUrls.split(',')
+        for (const url of urls) {
+            const client = ThorClient.fromUrl(url)
+            const genesis = await client.blocks.getGenesisBlock()
+            genesisIds.push(genesis!.id)
+        }
+
+        // Ensure all genesis blocks match
+        if (new Set(genesisIds).size !== 1) {
+            console.error('Genesis blocks do not match')
```

Comment:
I think we can add check if existing genesisid match the one we received from network

Reply:
already implemented https://github.com/vechain/thor-e2e-tests/blob/5208fed724c36bd87fbe8e0b65e27eac34d1d245/src/test-env.ts#L71